template-compose/domain/src/main/java/co/nimblehq/template/compose/domain/repository/AppPreferencesRepository.kt

```diff
@@ -0,0 +1,10 @@
+package co.nimblehq.template.compose.domain.repository
+
+import kotlinx.coroutines.flow.Flow
+
+interface AppPreferencesRepository {
+
+    fun getAppPreferences(): Flow<Boolean>
+
+    suspend fun updateAppPreferences(appPreferencesValue: Boolean)
```

Comment:
I think the base name (as the sample) could be just like this

```suggestion
    fun getPreference(): Flow<Boolean>

    suspend fun updatePreference(appPreferenceValue: Boolean)
```

Get or update is an action on a `single preference`, not on all preferences or appPreferences.